### PR TITLE
native: add address sanitizer make target

### PR DESCRIPTION
**Pull request by authmillenon**
Adds a make target to compile an application with the GCC address sanitizer.

### CODE REVIEW

**Comment by authmillenon** [comment]
Might be also of interest for @LudwigOrtmann and @BytesGalore to look at.

File: cpu/native/syscalls.c [hunk 1]
```diff
@@ -162,7 +166,7 @@ void *calloc(size_t nmemb, size_t size)
         }
         else {
             _native_in_calloc = 1;
-            *(void **)(&real_calloc) = dlsym(RTLD_NEXT, "calloc");
+            /* *(void **)(&real_calloc) = dlsym(RTLD_NEXT, "calloc"); */
```
Comment (OlegHahm) [inline cpu/native/syscalls.c]: Please explain why this is commented out and why it isn't removed?

**Comment by authmillenon** [comment]
Somehow the debugging symbols are missing for me… can somebody confirm? :/

File: cpu/native/syscalls.c (hunk 1 above)
Comment (authmillenon) [inline cpu/native/syscalls.c]: You were commenting to fast ;-)

**Comment by authmillenon** [comment]
Should I also add a target for the thread sanitizer?

**Comment by authmillenon** [comment]
> Somehow the debugging symbols are missing for me… can somebody confirm? :/

One needs a [symbolizer (like `llvm-symbolizer`) for that](http://stackoverflow.com/questions/21163828/meaningful-stack-traces-for-address-sanitizer-in-gcc). Sadly the ubuntu `llvm-symbolizer` already is differently called than in the referred link (`llvm-symbolizer-3.4`), so I need to think about how I will implement this. Any ideas are welcome

**Comment by authmillenon** [comment]
Ahh there is a fix in [GCC 4.9](http://tsdgeos.blogspot.de/2014/03/asan-and-gcc-how-to-get-line-numbers-in.html?showComment=1395330478558#c4687335237328570311) for that.

**Comment by authmillenon** [comment]
I'll document it in the wiki, as soon as this get's merged.

**Comment by PeterKietzmann** [comment]
IIRC there is a problem with the thread system for 32 bit compiling. Will look at that again.

**Comment by authmillenon** [comment]
You need to install libtsan:i386 probably...

File: boards/native/Makefile.include [hunk 2]
```diff
@@ -90,6 +90,11 @@ all-debug: export CFLAGS += -g
 all-cachegrind: export CFLAGS += -g
 all-gprof: export CFLAGS += -pg
 all-gprof: export LINKFLAGS += -pg
+all-asan: export CFLAGS += -fsanitize=address -fno-omit-frame-pointer -g
+all-asan: export CFLAGS += -DNATIVE_IN_CALLOC
+all-asan: export LINKFLAGS += -fsanitize=address -fno-omit-frame-pointer -g
+all-asan: export CFLAGS += -fsanitize=thread -fno-omit-frame-pointer -g
```
Comment (PeterKietzmann) [inline boards/native/Makefile.include]: I think you mean all- t san

**Comment by authmillenon** [comment]
You're right: there is no i386 support for libtsan in Debian and co. at least.

**Comment by PeterKietzmann** [comment]
Ahja, that was it :-) ! Please squash. Would be nice to hear @LudwigOrtmann commenting this. Otherwise I tend to merge this soon.

**Comment by PeterKietzmann** [comment]
@authmillenon please squash

**Comment by authmillenon** [comment]
Done.

**Comment by PeterKietzmann** [comment]
@LudwigOrtmann I'll merge this now as it worked for me and Travis. Still it would be nice if you have a quick look at this

**Comment by LudwigOrtmann** [comment]
@PeterKietzmann what exactly would you like my comment on?

**Comment by PeterKietzmann** [comment]
I just wanted you to look over this PR, what you obviously did. So I'm fine

**Pull request merged**